components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/CombinedHistorySuggestionProvider.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.awesomebar.provider
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import mozilla.components.browser.icons.BrowserIcons
+import mozilla.components.concept.awesomebar.AwesomeBar
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.storage.HistoryMetadata
+import mozilla.components.concept.storage.HistoryMetadataStorage
+import mozilla.components.concept.storage.HistoryStorage
+import mozilla.components.feature.session.SessionUseCases
+import java.util.UUID
+
+/**
+ * Return 5 history suggestions by default.
+ */
+const val DEFAULT_COMBINED_SUGGESTION_LIMIT = 5
+
+/**
+ * A [AwesomeBar.SuggestionProvider] implementation that combines suggestions from
+ * [HistoryMetadataSuggestionProvider] and [HistoryStorageSuggestionProvider].
+ * It will retrieve suggestions using [HistoryMetadataSuggestionProvider] first,
```

Comment:
nit: It will return, now that we retrieve concurrently.

Reply:
Thank you. Updated.